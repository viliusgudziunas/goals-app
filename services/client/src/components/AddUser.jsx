import React, { useState } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

const AddUser = ({ pingFetchUsers, setPingFetchUsers }) => {
  const [email, setEmail] = useState('');

  const handleFormSubmit = e => {
    e.preventDefault();
    const data = { email };
    axios
      .post(`${process.env.REACT_APP_USERS_SERVICE_URL}/users`, data)
      .then(() => {
        setPingFetchUsers(!pingFetchUsers);
        setEmail('');
      })
      .catch(err => {
        console.log(err);
      });
  };

  return (
    <div className='row'>
      <form className='col-md-4' onSubmit={handleFormSubmit}>
        <div className='form-group'>
          <input
            type='email'
            className='form-control'
            placeholder='Enter an email address'
            id='formEmail'
            value={email}
            onChange={e => setEmail(e.target.value)}
            required
          />
        </div>
        <button type='submit' className='btn btn-success btn-block'>
          Submit
        </button>
      </form>
    </div>
  );
};

AddUser.propTypes = {
  pingFetchUsers: PropTypes.bool.isRequired,
  setPingFetchUsers: PropTypes.func.isRequired
};

export default AddUser;
