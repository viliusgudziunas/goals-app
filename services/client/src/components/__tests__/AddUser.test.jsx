import React from 'react';
import { shallow } from 'enzyme';

import AddUser from '../AddUser';

const pingFetchUsers = true;
const setPingFetchUsers = jest.fn();

describe('<AddUser />', () => {
  const wrapper = shallow(
    <AddUser
      pingFetchUsers={pingFetchUsers}
      setPingFetchUsers={setPingFetchUsers}
    />
  );
  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  const form = wrapper.find('form');
  it('should have a form', () => {
    expect(form.length).toBe(1);
  });

  const formFields = form.find('input');
  it('should have one input field', () => {
    expect(formFields.length).toBe(1);
  });

  it('should have an email field', () => {
    expect(formFields.get(0).props.type).toBe('email');
  });

  it('should have a submit button', () => {
    expect(form.find('button').length).toBe(1);
    expect(form.find('button').get(0).props.type).toBe('submit');
  });
});
