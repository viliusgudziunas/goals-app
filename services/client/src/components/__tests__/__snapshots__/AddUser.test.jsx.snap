// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddUser /> should match the snapshot 1`] = `"<div class=\\"row\\"><form class=\\"col-md-4\\"><div class=\\"form-group\\"><input type=\\"email\\" class=\\"form-control\\" placeholder=\\"Enter an email address\\" id=\\"formEmail\\" value=\\"\\" required=\\"\\"/></div><button type=\\"submit\\" class=\\"btn btn-success btn-block\\">Submit</button></form></div>"`;
